File: July Lunchtime 2021 Contest/Red Alert.cpp
#include<iostream>
using namespace std;
int main(void)
{
	int n;
	cin>>n;
	while(n--){
		int x,d,h,sum=0;
		cin>>x>>d>>h;
		int a[x];
		for(int i=0;i<x;i++){
           cin>>a[i];   
			if(a[i]>0){
				sum+=a[i];
			}
			else{
				if(sum<d){
				
				  sum=0;
				  
			}
			else{
			
				sum=sum-d;
			}
   }
}
		if(sum<=h){
				cout<<"NO"<<"\n";
			}
			else{
				cout<<"YES"<<"\n";
			}
	}
}

File: July Lunchtime 2021 Contest/red alert.py
# cook your dish here
try:
    n=int(input())
    for i in range(n):
        x,d,h=input().split(' ')
        x=int(x)
        d=int(d)
        h=int(h)
        a = list(map(int, input().split( )))
        sum = 0
        for j in range(len(a)):
            if(a[j]>0):
                sum+=a[j]
            else:
                if(sum<d):
                    sum=0
                else:
                    sum-=d
        
            if(sum>h):
                print("YES")
                break
        else:
            print("NO")
except:
    pass
